show/bgp_frr_v4.py

```diff
@@ -24,43 +24,93 @@ def bgp():
+                required=False)
+@click.option('--namespace',
+                '-n',
+                'namespace',
+                default=None,
+                type=str,
+                show_default=True,
+                help='Namespace name or all',
+                callback=multi_asic_util.multi_asic_namespace_validation_callback)
+def neighbors(ipaddress, info_type, namespace):
     """Show IP (IPv4) BGP neighbors"""
 
-    command = 'sudo vtysh -c "show ip bgp neighbor'
-
+    command = 'show ip bgp neighbor'
     if ipaddress is not None:
-        command += ' {}'.format(ipaddress)
+        if not bgp_util.is_ipv4_address(ipaddress):
+            ctx = click.get_current_context()
+            ctx.fail("{} is not valid ipv4 address\n".format(ipaddress))
+        try:
+            actual_namespace = bgp_util.get_namespace_for_bgp_neighbor(
+                ipaddress)
+            if namespace is not None and namespace != actual_namespace:
+                click.echo(
+                    "[WARNING]: bgp neighbor {} is present in namespace {} not in {}"
+                    .format(ipaddress, actual_namespace, namespace))
 
-        # info_type is only valid if ipaddress is specified
-        if info_type is not None:
-            command += ' {}'.format(info_type)
+            # save the namespace in which the bgp neighbor is configured
+            namespace = actual_namespace
+
+            command += ' {}'.format(ipaddress)
 
-    command += '"'
+            # info_type is only valid if ipaddress is specified
+            if info_type is not None:
+                command += ' {}'.format(info_type)
+        except ValueError as err:
+            ctx = click.get_current_context()
+            ctx.fail("{}\n".format(err))
+
+    ns_list = multi_asic.get_namespace_list(namespace)
+    output = ""
+    for ns in ns_list:
+        output += bgp_util.run_bgp_command(command, ns)
+
+    click.echo(output.rstrip('\n'))
 
-    run_command(command)
 
 # 'network' subcommand ("show ip bgp network")
 @bgp.command()
-@click.argument('ipaddress', metavar='[<ipv4-address>|<ipv4-prefix>]', required=False)
-@click.argument('info_type', metavar='[bestpath|json|longer-prefixes|multipath]',
-                type=click.Choice(['bestpath', 'json', 'longer-prefixes', 'multipath']), required=False)
-def network(ipaddress, info_type):
+@click.argument('ipaddress',
+                metavar='[<ipv4-address>|<ipv4-prefix>]',
+                required=False)
+@click.argument('info_type',
+                metavar='[bestpath|json|longer-prefixes|multipath]',
+                type=click.Choice(
+                    ['bestpath', 'json', 'longer-prefixes', 'multipath']),
+                required=False)
+@click.option('--namespace',
+                '-n',
+                'namespace',
+                type=str,
+                show_default=True,
+                required=True if multi_asic.is_multi_asic is True else False,
```

Comment:
on mulit ASIC "namespace" is mandatory.  @gechiang What's the approach 'show ip route' takes? Does it also require namespace to be mandatory.

Reply:
I guess it's fine, it makes sense to print 'network' from a particular BGP instance.

Reply:
Each BGP instance in multi asic platform will have all the learnt prefixes. This command will present the information from one specific instance.